compiler/noirc_frontend/src/hir/comptime/errors.rs

```diff
@@ -5,7 +5,7 @@ use noirc_errors::{CustomDiagnostic, Location};
 use super::value::Value;
 
 /// The possible errors that can halt the interpreter.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
We don't actually use this trait on `TypeCheckError` (which causes the requirement for it here). We could just remove it from the other errors for now.